solidserver/data_source_ip_address.go

```diff
@@ -0,0 +1,146 @@
+package solidserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math"
+	"net/url"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceipaddress() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceipaddressRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeInt,
+				Description: "The id of the ip address",
+				Required:    true,
+			},
+			"space": {
+				Type:        schema.TypeString,
+				Description: "The name of the space into which creating the IP address.",
+				Computed:    true,
+			},
+			"subnet": {
+				Type:        schema.TypeString,
+				Description: "The name of the subnet into which creating the IP address.",
+				Computed:    true,
+			},
+			"pool": {
+				Type:        schema.TypeString,
+				Description: "The name of the pool into which creating the IP address.",
+				Computed:    true,
+			},
+			"address": {
+				Type:        schema.TypeString,
+				Description: "The provisionned IP address.",
+				Computed:    true,
+			},
+			"device": {
+				Type:        schema.TypeString,
+				Description: "Device Name to associate with the IP address (Require a 'Device Manager' license).",
+				Computed:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The long name or FQDN of the IP address to create.",
+				Computed:    true,
+			},
+			"mac": {
+				Type:        schema.TypeString,
+				Description: "The MAC Address of the IP address to create.",
+				Computed:    true,
+			},
+
+			"class": {
+				Type:        schema.TypeString,
+				Description: "The class associated to the IP address.",
+				Computed:    true,
+			},
+			"cidr": {
+				Type:        schema.TypeInt,
+				Description: "The CIDR associated to the IP address.",
+				Computed:    true,
+			},
+			"class_parameters": {
+				Type:        schema.TypeMap,
+				Description: "The class parameters associated to the IP address.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceipaddressRead(d *schema.ResourceData, meta interface{}) error {
+	s := meta.(*SOLIDserver)
+
+	parameters := url.Values{}
+	parameters.Add("ip_id", d.Id())
```

Comment:
Using the internal ID to find the IP address seems a bit disturbing. This could be done using the Space and IP address itself, in order to retrieve both the ID and meta-data. Would this make sense to you ?

Reply:
The goal of datasource in our case is to get the IP name after created it. We took the shortest path. For a generic usage of datasource, it looks to be a better way to use space and IP